xc7/make/arch_define.cmake

```diff
@@ -0,0 +1,77 @@
+function(ADD_ARCH_DEFINE)
+  set(options)
+  set(oneValueArgs ARCH ROI_PART ROI_DIR YOSYS_SCRIPT)
+  set(multiValueArgs)
+  cmake_parse_arguments(
+    ADD_ARCH_DEFINE
+    "${options}"
+    "${oneValueArgs}"
+    "${multiValueArgs}"
+    ${ARGN}
+  )
+
+  set(ARCH ${ADD_ARCH_DEFINE_ARCH})
+  set(ROI_PART ${ADD_ARCH_DEFINE_ROI_PART})
+  set(ROI_DIR ${ADD_ARCH_DEFINE_ROI_DIR})
+  set(YOSYS_SCRIPT ${ADD_ARCH_DEFINE_YOSYS_SCRIPT})
+
+  define_arch(
+	  ARCH ${ARCH}
+    # -flatten is used to ensure that the output eblif has only one module.
```

Comment:
This comment belongs with the YOSYS_SCRIPT definition.

Reply:
Ok, I will move it to where YOSYS_SCRIPT is defined